vllm/core/scheduler.py

```diff
@@ -510,8 +537,12 @@ def _schedule_running(
         # NOTE(woosuk): Preemption happens only when there is no available slot
         # to keep all the sequence groups in the RUNNING state.
 
-        running_queue = self.running
+        # Store original running requests for the case of async + preemption
+        if self.use_async_output_proc:
+            orig_running = self.running.copy()
```

Comment:
QQ: Is this cheap?

Reply:
yeah, since this is "shallow copy", only pointers are copied